### DISCUSSION TITLE
choose default text color value for HTML content

### DISCUSSION THREAD

**Discussion by nnako** [freeplane/freeplane#1729] [Help]
back in the days...

I think it was so with Freeplane v1.3.15, when selecting a text portion in the node core and then pressing `<CTRL> + <E>`, resulted in a foreground color change to `BLUE` of the selected text portion. The color value could be checked within the HTML editor (to be activated via `<ALT> + <RETURN>` and then selecting `HTML` on Windows computers) as follows:

```html
  <body>
    <p>
      <font color="#0000ff">test</font>
    </p>
  </body>
```

Now, e.g. with Freeplane v1.11.7 and later, the key binding for applying `BLUE` foreground color to the selected text changed from `<CTRL> + <E>` to `<CTRL> + <L>` and, more importantly, the default `BLUE` color value changed as well:

```html
  <body>
    <p>
      <span style="color: #0000c0;">test</span>
    </p>
  </body>
```

So, the default color value for `BLUE` changed from `#0000ff` to `#0000c0`.

Is it possible to set the default color values via some configuration / option back to `#0000ff` ? Or can this only be done at compile-time?

Thanks.

**Comment by i-plasm** [comment]
Hi @nnako , the colors are hardcoded in the SHTML library (edit: the specific shade depends on the caret color of the particular mindmap). However, you can use this script for changing the font color in core/details/note. The script requires that you assign and invoke it via a shortcut. Also, the script will not work on the floating external dialog that comes from an option like "Edit node core in dialog" - it works only on the embedded/in-line editors -:

``` groovy
import java.awt.Component
import java.awt.KeyboardFocusManager
import javax.swing.text.JTextComponent
import javax.swing.text.SimpleAttributeSet
import javax.swing.text.html.CSS
import javax.swing.text.html.HTML
import javax.swing.text.html.HTMLDocument
import javax.swing.text.html.StyleSheet

changeFonttHTML("#0000ff")

def changeFonttHTML(String color) {
  Component keyboardFocusOwner =
      KeyboardFocusManager.getCurrentKeyboardFocusManager().getFocusOwner()

  if (!(keyboardFocusOwner instanceof JTextComponent)) {
    return
  }

  JTextComponent editor = (JTextComponent) keyboardFocusOwner

  if (!editor.isEditable() || !(editor.getDocument() instanceof HTMLDocument)) {
    return
  }

  HTMLDocument doc = (HTMLDocument) editor.getDocument()

  StyleSheet styleSheet = doc.getStyleSheet()
  SimpleAttributeSet set = new SimpleAttributeSet()
  styleSheet.addCSSAttribute(set, CSS.Attribute.COLOR, color)
  set.addAttribute(HTML.Attribute.COLOR, color)
  doc.setCharacterAttributes(editor.getSelectionStart(),
      editor.getSelectionEnd() - editor.getSelectionStart(), set, false)
}
```

**Comment by macmarrum** [comment]
I think that the color (blue in this case), or rather its shade, depends on the original text color.
In case of a dark original text, dark blue is used, and when the original text is bright, light blue is used.

**Reply by macmarrum** [reply]
I think this is the exact logic, if you're interested:

`com.lightdev.app.shtm.SHTMLEditorKitActions.FixedFontColorAction`

```java
private Color getColorCloserToCaretColor() {
	Color color = panel.getEditorPane().getCaretColor();
	if(isDark(color))
		return darkColor;
	else
		return lightColor;
}

private boolean isDark(Color color) {
	int r = color.getRed();
	int g = color.getGreen();
	int b = color.getBlue();

	double brightness = Math.sqrt(
			  0.299 * (r * r) +
			  0.587 * (g * g) +
			  0.114 * (b * b)
			  );

          return brightness <= 127.5;
}
```
